src/amd.c

```diff
@@ -0,0 +1,109 @@
+/*
+ * AMD LED control
+ * Copyright (C) 2019, Advanced Micro Devices, Inc.
+ *
+ * This program is free software; you can redistribute it and/or modify it
+ * under the terms and conditions of the GNU General Public License,
+ * version 2, as published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+ * FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License for
+ * more details.
+ *
+ * You should have received a copy of the GNU General Public License along with
+ * this program; if not, write to the Free Software Foundation, Inc.,
+ * 51 Franklin St - Fifth Floor, Boston, MA 02110-1301 USA.
+ *
+ */
+
+#include <errno.h>
+#include <fcntl.h>
+#include <limits.h>
+#include <stdint.h>
+#include <stdio.h>
+#include <stdlib.h>
+#include <string.h>
+#include <unistd.h>
+#include <libgen.h>
+#include <inttypes.h>
+#include <sys/stat.h>
+#include <sys/mman.h>
+#include <sys/sysinfo.h>
+#include <sys/file.h>
+
+#if _HAVE_DMALLOC_H
+#include <dmalloc.h>
+#endif
+
+#include "config.h"
+#include "ibpi.h"
+#include "list.h"
+#include "utils.h"
+#include "amd.h"
+#include "amd_sgpio.h"
+
+int _find_file_path(const char *start_path, const char *filename,
+		    char *path, size_t path_len)
+{
+	int rc, found;
+	struct stat sbuf;
+	struct list dir;
+	char *dir_name;
+	const char *dir_path;
+
+	rc = scan_dir(start_path, &dir);
+	if (rc) {
+		log_info("Failed to scan %s", start_path);
+		return 0;
+	}
+
+	found = 0;
+	list_for_each(&dir, dir_path) {
+		dir_name = strrchr(dir_path, '/');
+		if (!dir_name++)
```

Comment:
If dir_name is NULL then increment causes undefined behavior. Please first check for NULL then increment pointer.